CMakeLists.txt

```diff
@@ -0,0 +1,177 @@
+cmake_minimum_required(VERSION 3.1)
+cmake_policy(SET CMP0074 NEW)
+
+include(${CMAKE_CURRENT_LIST_DIR}/cmake/Hunter/HunterGate.cmake)
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.23.179.tar.gz"
+    SHA1 "21b2f139feb9f3c37e367750013aee15e1c45355"
+)
+
+project(imgui VERSION 1.70)
+set(IMGUI_VERSION "1.70")
+
+if(MSVC)
+    #setup pdb info
+    set(pdb_output_dir "${CMAKE_CURRENT_BINARY_DIR}/pdb-files")
+
+    set(CMAKE_PDB_OUTPUT_DIRECTORY "${pdb_output_dir}")
+    set(CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY "${pdb_output_dir}") 
+
+    #need to sync pdp files
+    add_compile_options("/FS")
+endif()
+
+set(IMGUI_IMPL_DEFINES)
+set(IMGUI_IMPL_LIBRARIES)
+set(IMGUI_IMPL_EXAMPLES)
+
+option(IMGUI_IMPL_ALLEGRO5 "Include Allegro5 renderer" OFF)
+option(IMGUI_IMPL_MARMALADE "Include Marmalade renderer" OFF)
+option(IMGUI_IMPL_DIRECTX9 "Include DirectX 9 renderer" OFF)
+option(IMGUI_IMPL_DIRECTX10 "Include DirectX 10 renderer" OFF)
+option(IMGUI_IMPL_DIRECTX11 "Include DirectX 11 renderer" OFF)
+option(IMGUI_IMPL_DIRECTX12 "Include DirectX 12 renderer" OFF)
+option(IMGUI_IMPL_OPENGL2 "Include Opengl 2 renderer" OFF)
+option(IMGUI_IMPL_OPENGL3 "Include Opengl 3 renderer" OFF)
+option(IMGUI_IMPL_VULKAN "Include Vulkan renderer" OFF)
+option(IMGUI_IMPL_METAL "Include Metal renderer" OFF)
+
+option(IMGUI_IMPL_GLFW "Include GLFW window manager" OFF)
+option(IMGUI_IMPL_GLUT "Include GLUT window manager" OFF)
+option(IMGUI_IMPL_SDL "Include SDL window manager" OFF)
+option(IMGUI_IMPL_WIN32 "Include Win32 window manager" OFF)
+option(IMGUI_IMPL_OSX "Include OSX window manager" OFF)
+
+option(IMGUI_BUILD_EXAMPLES "Build imgui examples" OFF)
+option(IMGUI_ENABLE_TEST_ENGINE "Build imgui tests" OFF)
+
+set(IMGUI_HEADERS
+    imconfig.h
+    imgui.h
+    imgui_internal.h
+    imstb_rectpack.h
+    imstb_textedit.h
+    imstb_truetype.h
+)
+
+set(IMGUI_SOURCE
+    imgui.cpp
+    imgui_demo.cpp
+    imgui_draw.cpp
+    imgui_widgets.cpp
+)
+
+include(cmake/impl_allegro5.cmake)
+include(cmake/impl_marmalade.cmake)
+include(cmake/impl_directx9.cmake)
+include(cmake/impl_directx10.cmake)
+include(cmake/impl_directx11.cmake)
+include(cmake/impl_directx12.cmake)
+include(cmake/impl_opengl2.cmake)
+include(cmake/impl_opengl3.cmake)
+include(cmake/impl_vulkan.cmake)
+
+include(cmake/impl_glfw.cmake)
+include(cmake/impl_glut.cmake)
+include(cmake/impl_sdl.cmake)
+include(cmake/impl_win32.cmake)
+include(cmake/impl_osx.cmake)
+
+include_directories(${CMAKE_CURRENT_LIST_DIR})
+
+add_library(imgui STATIC ${IMGUI_HEADERS} ${IMGUI_SOURCE})
```

Comment:
Is there a reason to force `STATIC` here?

Reply:
I don't think the lib is completely setup for dll support, there is some things sprinkled in there but I don't know how ready it is.

Reply:
> I don't think the lib is completely setup for dll support

Can the [CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/latest/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html) help?

If you remove `STATIC` you can always force static build with `BUILD_SHARED_LIBS=OFF`.